corehq/apps/accounting/management/commands/fill_subscriptions.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+
+from django.core.management import BaseCommand
+from corehq.apps.accounting.models import SoftwarePlanEdition
+
+from corehq.apps.accounting.invoicing import should_create_invoice
+from corehq.apps.accounting.models import CustomerInvoice
+from six.moves import input
+
+from corehq.apps.accounting.invoicing import CustomerAccountInvoiceFactory
+
+
+class Command(BaseCommand):
+    help = 'find customer invoice without subscriptions details and correct the data'
+
+    def handle(self, **kwargs):
+        invoices = CustomerInvoice.objects.filter(subscriptions=None).all()
+        self.permit_invoice_to_change(invoices)
+
+        for invoice in invoices:
+            invoice_factory = CustomerAccountInvoiceFactory(
+                account=invoice.account,
+                date_start=invoice.date_start,
+                date_end=invoice.date_end
+            )
+            invoice_factory.create_invoice()
```

Comment:
Will this duplicate any pre-existing line items on these invoices, if they exist?

Reply:
No this wont. there is a check on the inside the genearete_invoice ufnction.

Reply:
it used get_or_create